File: easy/2748. Number of Beautiful Pairs/solution.java
class Solution {
    public int countBeautifulPairs(int[] nums) {
        int count = 0;
        int n = nums.length;
        
        for (int i = 0; i < n; i++) {
            for (int j = i + 1; j < n; j++) {
                int first = firstDigit(nums[i]);
                int last = nums[j] % 10;
                
                if (gcd(first, last) == 1) {
                    count++;
                }
            }
        }
        
        return count;
    }
    
    private int firstDigit(int num) {
        while (num >= 10) {
            num /= 10;
        }
        return num;
    }
    
    private int gcd(int a, int b) {
        if (b == 0) return a;
        return gcd(b, a % b);
    }
}

File: easy/2748. Number of Beautiful Pairs/solution.rs
impl Solution {
    pub fn count_beautiful_pairs(nums: Vec<i32>) -> i32 {
        fn gcd(a: i32, b: i32) -> i32 {
            if b == 0 { a } else { gcd(b, a % b) }
        }
        
        fn first_digit(mut num: i32) -> i32 {
            while num >= 10 {
                num /= 10;
            }
            num
        }
        
        let mut count = 0;
        let n = nums.len();
        
        for i in 0..n {
            for j in (i + 1)..n {
                let first = first_digit(nums[i]);
                let last = nums[j] % 10;
                
                if gcd(first, last) == 1 {
                    count += 1;
                }
            }
        }
        
        count
    }
}